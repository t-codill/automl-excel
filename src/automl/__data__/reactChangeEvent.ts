// tslint:disable: no-any
export const reactChangeEvent: React.ChangeEvent<any> = {
    target: {}
} as unknown as React.ChangeEvent<any>;
