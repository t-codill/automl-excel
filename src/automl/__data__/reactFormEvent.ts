// tslint:disable: no-any
export const reactFormEvent: React.FormEvent<any> = {
    preventDefault(): void { return; }
} as unknown as React.FormEvent<any>;
