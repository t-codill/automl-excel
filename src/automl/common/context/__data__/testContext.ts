import { PageNames } from "../../PageNames";
import { Logger } from "../../utils/logger";
import { IWorkspaceContextProps, WorkspaceError } from "../IWorkspaceProps";
import { WorkspaceFlight } from "../WorkspaceFlight";

export function getLogCustomEventSpy(): jest.SpyInstance<ReturnType<Logger["logCustomEvent"]>> {
    const spy = jest.spyOn(Logger.prototype, "logCustomEvent");
    spy.mockImplementation(jest.fn());
    return spy;
}
let pageName: PageNames = PageNames.Unknown;
export const testContext: IWorkspaceContextProps = {
    get pageName(): PageNames { return pageName; },
    get theme(): string { return "light"; },
    discoverUrls: {},
    flight: new WorkspaceFlight(""),
    location: "eastus",
    resourceGroupName: "testResource",
    subscriptionId: "00000000-0000-0000-0000-000000000000",
    logger: new Logger("test"),
    getToken: () => "testToken",
    setPageName: (value: PageNames) => { pageName = value; },
    workspaceName: "testWorkSpace",
    setNavigationBarButtons(): void { return; },
    onError: (err: WorkspaceError) => { throw err; },
    clearErrors: () => { return; },
    setHeader: () => { return; },
    setLoading: () => { return; }
};
