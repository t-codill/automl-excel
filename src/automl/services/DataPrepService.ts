import { AzureMachineLearningDataPrepClient, AzureMachineLearningDataPrepModels } from "@vienna/dataprep";
import { IServiceBaseProps } from "./ServiceBase";
import { ServiceBaseNonArm } from "./ServiceBaseNonArm";

export class DataPrepService extends ServiceBaseNonArm<AzureMachineLearningDataPrepClient> {
    constructor(props: IServiceBaseProps) {
        super(props, AzureMachineLearningDataPrepClient, props.discoverUrls.history);
    }

    public async startDataProfiling(
        dataStoreName: string,
        blobName: string,
        experimentName: string,
        computeTargetName: string,
        topNSample: number
    ): Promise<AzureMachineLearningDataPrepModels.ProfileJobDto | undefined> {

        const response = await this.send(async (client, abortSignal) => {
            return client.dataPrep.startProfileJob(this.props.subscriptionId, this.props.resourceGroupName, this.props.workspaceName, {
                abortSignal,
                dto: {
                    computeTargetName,
                    dataStoreName,
                    dataPath: blobName,
                    experimentName,
                    topNSample,
                    outputVersion: "2.0"
                }
            });
        });
        return response;
    }
}
