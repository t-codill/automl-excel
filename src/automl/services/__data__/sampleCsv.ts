export const sampleCsv = `a,b,c,d,e
1,2,3,4,5
1,2,3,4,5
1,2,3,4,5
1,2,3,4,5
1,2,3,4,5
`;

const largeCsvHeader = Array(50)
    .fill("aaaaa,bbbbb,ccccc,ddddd,eeeee")
    .join(",");

const largeCsvRow = Array(50)
    .fill("11111,22222,33333,44444,55555")
    .join(",");

export const largeCsv = `${largeCsvHeader}
${Array(50)
        .fill(largeCsvRow)
        .join("\n")}
`;
