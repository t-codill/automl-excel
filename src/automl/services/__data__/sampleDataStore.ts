import { AzureMachineLearningDatastoreManagementModels } from "@vienna/datastore";

export const sampleDataStore: Required<AzureMachineLearningDatastoreManagementModels.DataStoreDto> = {
    dataStoreType: "AzureBlob",
    name: "defaultDataStore",
    azureStorageSection: {
        accountKey: "key",
        accountName: "account",
        credentialType: "AccountKey",
        credential: "secret",
        containerName: "defaultContainer",
        isSas: false
    },
    azureDataLakeSection: {},
    azurePostgreSqlSection: {},
    azureSqlDatabaseSection: {},
    hasBeenValidated: true
};
