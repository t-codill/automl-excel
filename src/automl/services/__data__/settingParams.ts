import { AdvancedSetting } from "../AdvancedSetting";

export const settingParams: AdvancedSetting = {
    jobType: "classification",
    column: "1",
    metric: "AUC_weighted",
    experimentTimeoutMinutes: 60,
    maxIteration: 15,
    experimentExitScore: 0.99,
    maxConcurrent: 1,
    maxCores: 2,
    preprocess: true,
    nCrossValidations: 5,
    blacklistAlgos: null,
    validationSize: 0.2,
    grainColumns: null,
    maxHorizon: null,
    timeSeriesColumn: null
};
