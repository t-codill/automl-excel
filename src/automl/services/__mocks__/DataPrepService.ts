import { AzureMachineLearningDataPrepModels } from "@vienna/dataprep";

export class DataPrepService {
    private readonly dataPath = "dataPath";
    private readonly profileResultPath = "profileResultPath";
    private readonly runId = "runId";

    public async startDataProfiling(
        dataStoreName: string,
        _blobName: string,
        experimentName: string,
        computeTargetName: string,
        _topNSample: number
    ): Promise<AzureMachineLearningDataPrepModels.ProfileJobDto | undefined> {
        return {
            computeTargetName,
            dataStoreName,
            experimentName,
            dataPath: this.dataPath,
            profileResultPath: this.profileResultPath,
            runId: this.runId
        };
    }

    public dispose(): void {
        return;
    }

    public reset(): void {
        return;
    }
}
