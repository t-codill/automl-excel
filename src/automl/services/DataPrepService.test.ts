import { testContext } from "../common/context/__data__/testContext";
import { DataPrepService } from "./DataPrepService";

jest.mock("@vienna/dataprep");

const service: DataPrepService = new DataPrepService(testContext);

describe("DataPrepService", () => {
    it("should start", async () => {
        const result = await service.startDataProfiling("dataStoreName",
            "blobName",
            "experimentName",
            "computeTargetName",
            10000);
        expect(result)
            .toMatchSnapshot();
    });
});
