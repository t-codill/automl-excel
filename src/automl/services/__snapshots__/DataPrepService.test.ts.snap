// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataPrepService should start 1`] = `
Object {
  "_mockParameters": Array [
    "00000000-0000-0000-0000-000000000000",
    "testResource",
    "testWorkSpace",
    Object {
      "abortSignal": Object {},
      "dto": Object {
        "computeTargetName": "computeTargetName",
        "dataPath": "blobName",
        "dataStoreName": "dataStoreName",
        "experimentName": "experimentName",
        "outputVersion": "2.0",
        "topNSample": 10000,
      },
    },
  ],
  "dataPath": "dataPath",
  "profileResultPath": "profileResultPath",
  "runId": "runId",
}
`;
